modules/integration/tests-ui-integration/src/test/java/org/wso2/am/integration/ui/tests/APIMANAGER3250CrossTenantSubscriptionTestCase.java

```diff
@@ -0,0 +1,225 @@
+        // wait until load the page
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("username")));
+        driver.findElement(By.id("username")).clear();
+        driver.findElement(By.id("username")).sendKeys("admin@test1.com");
+        driver.findElement(By.id("pass")).clear();
+        driver.findElement(By.id("pass")).sendKeys(TEST_DATA_PASSWORD);
+        driver.findElement(By.id("loginButton")).click();
+        // wait until load the page
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.linkText("Add")));
+        // create new API
+        driver.findElement(By.linkText("Add")).click();
+        driver.findElement(By.id("name")).clear();
+        driver.findElement(By.id("name")).sendKeys(TEST_DATA_API_NAME);
+        driver.findElement(By.id("context")).clear();
+        driver.findElement(By.id("context")).sendKeys(TEST_DATA_API_NAME);
+        driver.findElement(By.id("version")).clear();
+        driver.findElement(By.id("version")).sendKeys(TEST_DATA_API_VERSION);
+
+        driver.findElement(By.id("go_to_implement")).click();
+
+        driver.findElement(By.cssSelector("a.btn:nth-child(4)")).click();
+
+
+        driver.findElement(By.id("jsonform-0-elt-production_endpoints")).clear();
+        driver.findElement(By.id("jsonform-0-elt-production_endpoints")).sendKeys(TEST_DATA_API_END_POINT);
+
+        driver.findElement(By.id("go_to_manage")).click();
+
+        driver.findElement(By.cssSelector("button.multiselect")).click();
+        driver.findElement(By.cssSelector(
+                ".multiselect-container > li:nth-child(2) > a:nth-child(1) > label:nth-child(1) > input:nth-child(1)"))
+                .click();
+        new Select(driver.findElement(By.id("subscriptions"))).selectByValue("all_tenants");
+        driver.findElement(By.id("publish_api")).click();
+
+        //check whether the publish is success
+        driver.findElement(By.id("lifecyclesLink")).click();
+
+        //browse store
+        driver.get(getStoreURL() + "?tenant=test2.com");
+
+        log.info("Started to Login to Store");
+        driver.findElement(By.id("login-link")).click();
+        WebElement userNameField = driver.findElement(By.id("username"));
+        WebElement passwordField = driver.findElement(By.id("password"));
+
+        userNameField.sendKeys("admin@test2.com");
+        passwordField.sendKeys(TEST_DATA_PASSWORD);
+        driver.findElement(By.id("loginBtn")).click();
+
+        //check the presence of admin name in store home page to verify the user has logged to store.
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.linkText("admin@test2.com")));
+        log.info("Logging to store is successful");
+
+        driver.get(getStoreURL() + "?tenant=test1.com");
+
+        //wait for few seconds and refresh the store since it will take little time to appear the published APIs in store
+        Thread.sleep(20000);
```

Comment:
Better to use proper wait function here. Don't use thread.sleeps unless it is really required.